### DISCUSSION TITLE
(Dayjs) imports in server directory

### DISCUSSION THREAD

**Discussion by lukas-schaetzle** [nuxt/nuxt#21791] [Questions]
I want to use the npm package [dayjs](https://day.js.org/) for a server api route. The main import (`import dayjs from "dayjs"`) works but to use dayjs plugins I also have to use imports with paths like this:
```js
import relativeTime from "dayjs/plugin/relativeTime"
import isBetween from "dayjs/plugin/isBetween"
import "dayjs/locale/de"
```

Unfortunately this does not work in the server directory. It always results in module not found errors like this:
```
Cannot find module '/Users/.../project-folder/.output/server/node_modules/dayjs/plugin/relativeTime' imported from /Users/.../project-folder/.output/server/chunks/api-test.post.mjs
Did you mean to import dayjs/plugin/relativeTime.js?
```

Prefixing the paths with `~/node_modules/` did also not work. And the currently maintained dayjs plugin fur nuxt 3 does also not work in server directories: see https://github.com/fumeapp/dayjs/issues/8

How can one use such imports (or specifically dayjs plugins) in the server directory?

**Comment by maou-shonen** [comment]
For dayjs, I'm using another module,
However, the plugins they register are all global
You should be able to register them in the `plugins` options of the module.

**Reply by lukas-schaetzle** [reply]
So which module are you using then? For the mentioned one I know that I can register plugins globally in the nuxt.config.ts via the `plugins` keyword. However, this question is not about how I can register plugins for the dayjs module but how I can use dayjs or any other path import in general in nuxt's server directory. (As said before the mentioned module is not working and importing it manually along with its plugins did also not work.)

**Reply by maou-shonen** [reply]
Oh, I was mistaken.
Indeed, it cannot be used directly on the server,
It needs to be registered in `auto imports` and used with ES transform.
Perhaps in a few days, I can provide a PR. (if I have some free time)

**Reply by lukas-schaetzle** [reply]
That would be great :)

**Comment by lukas-schaetzle** [comment]
Fixed it by adding this to the `nuxt.config.ts`:
```
  build: {
    transpile:
      process.env.NODE_ENV === 'production'
        ? ['dayjs']
        : []
  }
```

Also the language plugin has to be loaded via the default import instead of the side effect import to work correctly in the built server:
```javascript
import locale_de from 'dayjs/locale/de'
dayjs.locale(locale_de)

/*
   instead of:
   import 'dayjs/locale/de'
   dayjs.locale('de')
*/
```

**Comment by MinSomai** [comment]
```js
  build: {
    transpile: lifecycle === "build" ? ["ant-design-vue", "compute-scroll-into-view", "swiper", "dayjs"] : [],
  },

```

Nuxt3 ^
